packages/frames.js/src/core/utils.ts

```diff
@@ -24,65 +25,97 @@ function isValidButtonAction(action: any): action is "post" | "post_redirect" {
   );
 }
 
-/**
- * This function generates fully qualified URL for post and post_redirect buttons
- * that also encodes the button type and button value so we can use that on next frame.
- */
-export function generatePostButtonTargetURL({
-  buttonIndex,
-  buttonAction,
+export function generateTargetURL({
   currentURL,
   target,
   basePath,
-  state,
 }: {
-  buttonIndex: 1 | 2 | 3 | 4;
-  buttonAction: "post" | "post_redirect";
   currentURL: URL;
   basePath: string;
-  target: string | undefined;
-  /**
-   * Button state
-   */
-  state: JsonValue | undefined;
+  target: string | UrlObject | undefined;
 }): string {
   let url = new URL(currentURL);
 
-  if (target) {
+  if (
+    target &&
+    typeof target == "string" &&
+    (target.startsWith("http://") || target.startsWith("https://"))
```

Comment:
this breaks for mint buttons right?

Reply:
https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-10.md you can construct there urls using `new URL('polkadot:b0a8d493285c2df73290dfb7e61f870f:5hmuyxw9xdgbpptgypokw4thfyoe3ryenebr381z9iaegmfy')`.